ndk-build/src/apk.rs

```diff
@@ -79,17 +79,13 @@ impl<'a> UnalignedApk<'a> {
             return Err(NdkError::PathNotFound(path.into()));
         }
         let abi = target.android_abi();
-        let file_name = path.file_name().unwrap();
-        let out = self.0.build_dir.join("lib").join(abi);
-        std::fs::create_dir_all(&out)?;
-        std::fs::copy(path, out.join(&file_name))?;
+        let lib_path = Path::new("lib").join(abi).join(path.file_name().unwrap());
+        let out = self.0.build_dir.join(&lib_path);
+        std::fs::create_dir_all(out.parent().unwrap())?;
+        std::fs::copy(path, out)?;
 
         let mut aapt = self.0.build_tool(bin!("aapt"))?;
-        aapt.arg("add").arg(self.0.unaligned_apk()).arg(format!(
-            "lib/{}/{}",
-            abi,
-            file_name.to_str().unwrap()
-        ));
+        aapt.arg("add").arg(self.0.unaligned_apk()).arg(&lib_path);
```

Comment:
Unfortunately, this results in issues when building on Windows..
Using a backslash path for the library results in a runtime error where it fails to load the .so file.
```
prev: "lib/arm64-v8a/libhello_world.so"
new: "lib\\arm64-v8a\\libhello_world.so"
```

This only applies to the `lib_path`, for the `unaligned_apk` path it doesn't matter

Reply:
@msiglreith Ouch, small cleanups always result in horrible fallout. We need to resurrect our Windows CI (if this was ever tested on Windows at all). We've seen this before and I'm thinking of fixing it just like https://github.com/dvc94ch/cargo-subcommand/pull/1, will also take a look at the CI. Thanks for mentioning!

Reply:
https://github.com/rust-windowing/android-ndk-rs/runs/3165210588 So the CI runs fine, but the backslashes are now part of a full filename instead of serving as path separators.

Reply:
Hmm, this would need to run inside the emulator I guess. As the runtime crash is on the target device not during building.

Example:
```
AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{rust.wgpu_android/android.app.NativeActivity}: java.lang.IllegalArgumentException: Unable to find native library wgpu_android using classloader: dalvik.system.PathClassLoader[DexPathList[[],nativeLibraryDirectories=[/data/app/rust.wgpu_android-pSe8ljSIP942rgeN5FSlQQ==/lib/arm64, /system/lib64, /system/product/lib64]]]
```

Reply:
Correct, I think that's a good idea. Can we make this use artifacts from `cargo apk build`? That finally gives us a nice dependency graph in the Actions run overview, though I don't know if those can be shared between yml "files", which seem to be rendered as separate graphs (apparently called "runs"). We can just combine the two jobs in one yml file/run.

Reply:
Yeah, workflow sharing is not possible afaik - we would need to define it as action and then transfer the artifacts inside the same workflow, but I never tried that :D